src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -615,6 +648,100 @@ const getSshInfo = async (frameworkName) => {
   throw createError('Not Found', 'NoJobSshInfoError', `SSH info of job ${frameworkName} is not found.`);
 };
 
+const generateExitDiagnostics = (diag) => {
+  const podCompletionStatus = extractPodCompletionStatus(diag);
```

Comment:
Why not directly use diag?

Reply:
Since message in diag is messy。 It container string such as: "userLogs: \"logs\"\\n". multi escape character is used. 
If we use it directly, we need to change web portal to make this message show neatly.
Since this filed just a string and we don't promise any format here. Try to format this string in rest-server for backward compatibility

Reply:
If so, can we also merge generateExitDiagnostics with extractXXX? They depends on each other and reduce the match times.

---
In reply to: [321640867](https://github.com/microsoft/pai/pull/3575#discussion_r321640867) [](ancestors = 321640867)